tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -40,6 +187,98 @@ def check_interfaces_up(duthost, namespace, up_ports):
         return False
 
 
+def get_namespace_cmd_option(duthost, asic_index):
+    """Get the namespace option used in the command"""
+    namespace = duthost.get_namespace_from_asic_id(asic_index)
+    return "-n {}".format(namespace) if namespace else ""
+
+
+def get_down_ports(duthost, ports):
+    """Check and return the down ports among the given ports."""
+    return duthost.show_interface(command="status", up_ports=ports)["ansible_facts"][
+        "ansible_interface_link_down_ports"]
+
+
+def is_interface_status_expected(duthost, ports, expect_up=True):
+    """Check if the given ports are up or down as expected."""
+    if expect_up:
+        return len(get_down_ports(duthost, ports)) == 0
+    else:
+        return len(get_down_ports(duthost, ports)) == len(ports)
+
+
+def check_interface_status(duthost, ports, expect_up=True, wait_time=None):
+    """
+    Check if the given ports are up or down as expected.
+
+    Args:
+        duthost: DUT host object
+        ports: List of ports to check
+        expect_up: True if the ports are expected to be up, False if down
+        wait_time: Time to wait for the ports to come up or down
+    """
+    expect_status_str = "up" if expect_up else "down"
+    err_msg = ""
+
+    if wait_time is None:
+        port_down_wait_time, port_up_wait_time = \
+            default_port_toggle_wait_time(duthost, len(ports))
+        if expect_up:
+            wait_time = port_up_wait_time
+        else:
+            wait_time = port_down_wait_time
+
+    logging.info("Wait for ports to come {}: {}".format(expect_status_str, ports))
+    is_ok = wait_until(wait_time, 5, 0,
+                       is_interface_status_expected,
+                       duthost, ports, expect_up)
+
+    if not is_ok:
+        down_ports = get_down_ports(duthost, ports)
+        if expect_up:
+            problematic_ports = down_ports
+        else:
+            problematic_ports = set(ports) - down_ports
+
+        err_msg = "Some ports did not come {} as expected: {}".format(
+            expect_status_str, str(problematic_ports))
+    return is_ok, err_msg
+
+
+def get_phy_intfs_to_test_per_asic(duthost,
+                                   conn_graph_facts,
+                                   enum_frontend_asic_index,
+                                   xcvr_skip_list):
+    """
+    Get the interfaces to test for given asic, excluding the skipped ones.
+
+    return:
+        phy_intfs_to_test_per_asic: dict of all physical interfaces to test (key: physical port number,
+                                    value: list of logical interfaces under this physical port)
+    """
+    portmap, dev_conn = get_dev_conn(duthost,
```

Comment:
@longhuan-cisco Based on my understanding, the portmap here does not yield physical port numbers for device with breakout port. `portmap` here seems to be a dict of logical_port_name as key and an integer number starting from 0 with all logical ports in a sorted order.
https://github.com/sonic-net/sonic-mgmt/blob/e7f3a300c9f476ddc1d3a2fe51318b33d226f0ec/ansible/library/config_facts.py#L99-L100

I also tried executing your changes on a device with breakout ports and observed that every physical port has just 1 logical port. Are you observing similar behavior on a device with breakout ports too?

Due to the above issue, the DOM config command will fail for logical ports which have subport > 1.

Reply:
Right, today's portmap is not reliable for breakout case. 
Updated it to rely on get_physical_port_indices() function which is based on `index` from port config.

```
platform_tests/sfp/test_sfputil.py::test_check_sfputil_reset[mathilda-01-None] PASSED                              [100%]

  Ethernet8                 2320      25G   9100    N/A    etp2a  routed      up       up  QSFP28 or later         N/A
  Ethernet9                 2321      25G   9100    N/A    etp2b  routed      up       up  QSFP28 or later         N/A
 Ethernet10                 2322      25G   9100    N/A    etp2c  routed      up       up  QSFP28 or later         N/A
 Ethernet11                 2323      25G   9100    N/A    etp2d  routed      up       up  QSFP28 or later         N/A

09/09/2024 03:54:02 test_sfputil.get_phy_intfs_to_test_per_a L0304 INFO   | Interfaces to test for asic None: {2: [u'Ethernet8', u'Ethernet9', u'Ethernet10', u'Ethernet11']}

09/09/2024 03:54:02 test_sfputil.skip_lpmode_handle_for_phys L0176 INFO   | Skip lpmode handling for physical interface 2 as it's CMIS compliant(False) or Power Class 1(True)

09/09/2024 03:54:02 test_sfputil.disable                     L0080 INFO   | Disable DOM polling to avoid race condition during sfp reset for Ethernet8
09/09/2024 03:54:08 test_sfputil.disable                     L0089 INFO   | Shutdown Ethernet8 before sfp reset
09/09/2024 03:54:10 test_sfputil.disable                     L0089 INFO   | Shutdown Ethernet9 before sfp reset
09/09/2024 03:54:13 test_sfputil.disable                     L0089 INFO   | Shutdown Ethernet10 before sfp reset
09/09/2024 03:54:15 test_sfputil.disable                     L0089 INFO   | Shutdown Ethernet11 before sfp reset
09/09/2024 03:54:35 test_sfputil.restore                     L0124 INFO   | Startup Ethernet8 after sfp reset to restore module
09/09/2024 03:54:37 test_sfputil.restore                     L0130 INFO   | Restore DOM polling after sfp reset for Ethernet8
09/09/2024 03:54:38 test_sfputil.restore                     L0124 INFO   | Startup Ethernet9 after sfp reset to restore module
09/09/2024 03:54:40 test_sfputil.restore                     L0124 INFO   | Startup Ethernet10 after sfp reset to restore module
09/09/2024 03:54:42 test_sfputil.restore                     L0124 INFO   | Startup Ethernet11 after sfp reset to restore module
```